.github/workflows/helm-e2e-test.yml

```diff
@@ -0,0 +1,325 @@
+        env:
+          SAFE_REF_NAME: ${{ needs.prepare-environment.outputs.safe_ref_name }}
+        run: |
+          # Replace repository URLs in chart files
+          sed -i 's|ghcr\.io/nvidia/nvsentinel-|localhost/nvsentinel-e2e/nvsentinel-|g' \
+            distros/kubernetes/nvsentinel/charts/*/values.yaml \
+            distros/kubernetes/nvsentinel/values.yaml
+
+          # Create patched values file
+          cp tilt/release/values-release.yaml /tmp/values-patched.yaml
+          sed -i '/^global:/a\
+            image:\
+              tag: "'"${SAFE_REF_NAME}"'"\
+            imagePullSecrets: []' /tmp/values-patched.yaml
+
+          # Add MongoDB scheduling for real nodes only
+          cat >> /tmp/values-patched.yaml <<'EOF'
+          mongodb:
+            affinity:
+              nodeAffinity:
+                requiredDuringSchedulingIgnoredDuringExecution:
+                  nodeSelectorTerms:
+                  - matchExpressions:
+                    - key: type
+                      operator: NotIn
+                      values: ["kwok"]
+                    - key: node-role.kubernetes.io/control-plane
+                      operator: DoesNotExist
+            tolerations:
+            - operator: Exists
+          EOF
+
+      - name: Install NVSentinel via Helm
+        run: |
+          helm upgrade --install nvsentinel ./distros/kubernetes/nvsentinel \
+            --create-namespace \
+            --namespace ${{ env.NVSENTINEL_NAMESPACE }} \
+            --values /tmp/values-patched.yaml \
+            --set global.image.tag="${{ needs.prepare-environment.outputs.safe_ref_name }}" \
+            --debug
+
+          kubectl get pods -n ${{ env.NVSENTINEL_NAMESPACE }}
+
+      - name: Validate deployment with retry
+        env:
+          SAFE_REF_NAME: ${{ needs.prepare-environment.outputs.safe_ref_name }}
+        run: |
+          chmod +x scripts/validate-nvsentinel.sh
+          max_attempts=$((${{ env.VALIDATION_TIMEOUT_MINUTES }} * 60 / ${{ env.VALIDATION_INTERVAL_SECONDS }}))
+
+          for attempt in $(seq 1 $max_attempts); do
+            if ./scripts/validate-nvsentinel.sh --version "${SAFE_REF_NAME}" --namespace "${{ env.NVSENTINEL_NAMESPACE }}" --image-pattern "localhost/nvsentinel-e2e/nvsentinel" --verbose; then
+              echo "Validation passed on attempt $attempt"
+              exit 0
+            fi
+            [ $attempt -lt $max_attempts ] && sleep ${{ env.VALIDATION_INTERVAL_SECONDS }}
+          done
+
+          echo "Validation failed after $max_attempts attempts"
+          kubectl get pods -n ${{ env.NVSENTINEL_NAMESPACE }} -o wide
+          kubectl get events -n ${{ env.NVSENTINEL_NAMESPACE }} --sort-by='.lastTimestamp'
+          exit 1
+
+      - name: Export Kind logs
+        if: always()
+        run: |
+          mkdir -p /tmp/kind-logs
+          kind export logs /tmp/kind-logs --name "${CLUSTER_NAME}" || true
+
+      - name: Collect debug artifacts
```

Comment:
I like this, can we add this to the e2e tests as well? This will help debug test failures more easily

Reply:
YES! definitely will add it.